src/pow.cpp

```diff
@@ -79,6 +79,7 @@ unsigned int CalculateNextWorkRequired(const CBlockIndex* pindexLast, int64_t nF
     bnNew.SetCompact(pindexLast->nBits);
     bnOld = bnNew;
     bnNew *= nActualTimespan;
+    if (bnNew / nActualTimespan != bnOld) bnNew = bnPowLimit; else //MVF-BU Add overflow handle
```

Comment:
I agree that we should leave the original code for the pre-fork case, to remain as compatible with the rest of the network as possible.

However, we should ensure that post-fork, this overflow condition is fixed permanently.
We could have MVFisWithinRetargetPeriod() return true if Height > FinalActivateForkHeight , and remove the additional Height < MVFRetargetPeriodEnd() constraint.
Then the MVF functions would be in control forever post-fork, and the problem would be solved permanently.

What do you think about this?